### PR TITLE
Fix .NET Core project tests

### LINKED ISSUE #18996: CSharpErrorListDesktop.ErrorLevelWarning test and related are flakey
Roslyn.VisualStudio.IntegrationTests.CSharp.CSharpErrorListNetCore.ErrorList
Roslyn.VisualStudio.IntegrationTests.CSharp.CSharpErrorListNetCore.ErrorLevelWarning
Roslyn.VisualStudio.IntegrationTests.CSharp.CSharpErrorListDesktop.ErrorList
Roslyn.VisualStudio.IntegrationTests.CSharp.CSharpErrorListDesktop.ErrorLevelWarning

### LINKED ISSUE #19090: investigate flaky netcore tests

### LINKED ISSUE #19091: CSharpSquigglesDesktop and CSharpSquigglesNetCore tests are flaky
VerifySemanticErrorSquiggles
VerifySyntaxErrorSquiggles

### PR DESCRIPTION
**Pull request by sharwell**
Closes #18996
Closes #19090
Closes #19091

Implements a workaround for #32121.

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/ErrorList_InProc.cs [hunk 1]
```diff
@@ -32,73 +32,145 @@ public void WaitForNoErrorsInErrorList()
...
-            switch (errorLevel)
+            var errorItem = item as IVsErrorItem;
+            ErrorHandler.ThrowOnFailure(errorItem.GetHierarchy(out var hierarchy));
+            ErrorHandler.ThrowOnFailure(hierarchy.GetProperty((uint)VSConstants.VSITEMID.Root, (int)__VSHPROPID.VSHPROPID_Name, out var name));
+            return (string)name;
+        }
+
+        public static string GetFileName(this IVsTaskItem item)
+        {
+            ErrorHandler.ThrowOnFailure(item.Document(out var fileName));
```
Comment (mavasani) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/ErrorList_InProc.cs]: Does this return an empty string for error list items without document name?

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/SolutionExplorer_InProc.cs [hunk 2]
```diff
@@ -815,8 +926,11 @@ public void ReloadProject(string projectRelativePath)
             _solution.AddFromFile(projectPath);
         }
 
-        public void RestoreNuGetPackages()
-            => ExecuteCommand(WellKnownCommandNames.ProjectAndSolutionContextMenus_Solution_RestoreNuGetPackages);
+        public bool RestoreNuGetPackages(string projectName)
+        {
+            var solutionRestoreService = InvokeOnUIThread(() => GetComponentModel().GetExtensions<IVsSolutionRestoreService2>().Single());
```
Comment (mavasani) [inline src/VisualStudio/IntegrationTest/TestUtilities/InProcess/SolutionExplorer_InProc.cs]: Much more sane!

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/ErrorList_InProc.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/IntegrationTest/TestUtilities/InProcess/ErrorList_InProc.cs]: I'm not sure. It's the pattern I used in the test harness experiment and it worked for all the tests we have today. We can always adjust it if we need to test a new scenario _and_ we haven't completed the switch to Apex yet.

File: src/VisualStudio/IntegrationTest/TestUtilities/InProcess/ErrorList_InProc.cs [hunk 3]
```diff
@@ -32,73 +32,145 @@ public void WaitForNoErrorsInErrorList()
 
         public void NavigateToErrorListItem(int itemIndex)
         {
-            var errorItems = GetErrorItems().AsEnumerable();
+            var errorItems = GetErrorItems().AsEnumerable().ToArray();
             if (itemIndex > errorItems.Count())
```
[the remark this answers is not in the document]
Comment (sharwell) [reply src/VisualStudio/IntegrationTest/TestUtilities/InProcess/ErrorList_InProc.cs]: Agree, but I'm fine with leaving this so we can move the PR forward more efficiently and it's only test code.

**Comment by sharwell** [comment]
@jaredpar for approval

**Pull request merged**